ignite/cmd/model/chain_serve.go

```diff
@@ -0,0 +1,265 @@
+package cmdmodel
+
+import (
+	"fmt"
+	"strings"
+
+	tea "github.com/charmbracelet/bubbletea"
+
+	"github.com/ignite/cli/ignite/pkg/cliui/colors"
+	"github.com/ignite/cli/ignite/pkg/cliui/icons"
+	cliuimodel "github.com/ignite/cli/ignite/pkg/cliui/model"
+	"github.com/ignite/cli/ignite/pkg/cliui/style"
+	"github.com/ignite/cli/ignite/pkg/events"
+)
+
+const (
+	maxStatusEvents = 7
+)
+
+const (
+	stateChainServeStarting uint = iota
+	stateChainServeRunning
+	stateChainServeRebuilding
+	stateChainServeQuitting
+)
+
+var (
+	msgStopServe  = style.Faint.Render("Press the 'q' key to stop serve")
+	msgWaitingFix = colors.Info("Waiting for a fix before retrying...") // TODO: Replace colors by lipgloss styles
+)
+
+// NewChainServe returns a new UI model for the chain serve command.
+func NewChainServe(ctx Context, bus events.Provider, cmd tea.Cmd) ChainServe {
+	return ChainServe{
+		model:        newModel(ctx, cmd),
+		startModel:   cliuimodel.NewStatusEvents(bus, maxStatusEvents),
+		runModel:     cliuimodel.NewEvents(bus),
+		rebuildModel: cliuimodel.NewStatusEvents(bus, maxStatusEvents),
+		quitModel:    cliuimodel.NewEvents(bus),
+	}
+}
+
+// ChainServe defines a UI model for the chain serve command.
+type ChainServe struct {
+	model
+
+	state  uint  // Keeps track of the model/view being displayed
+	broken bool  // True when blockchain app's source code has issues
+	error  error // Critical error returned during command execution
```

Comment:
One my side I think we should completely remove `model`. 

Its role is to avoid repetition but actually any new model that embeds `model` have to repeat the same code 

- in `Init`:
```go
	return tea.Batch(m.startModel.WaitEvent, m.model.Init()) // <---
```

- in `Update`: 
```go
default:
		if m.model, cmd = m.model.Update(msg); cmd != nil {
			return m, cmd
		}
```

Repeating the 2 cases that handles the exit is more obvious/simple IMO. Plus for the moment we have only one model that embeds `model`.

Reply:
Yeah, I agree with this

Reply:
Changeset 02680fd